src/main/scala/com.snowplowanalytics.sauna/observers/Observer.scala

```diff
@@ -89,11 +108,29 @@ object Observer {
   /**
    * File has been published on AWS S3
    *
-   * @param path full path on S3 bucket
+   * @param id       full path on S3 bucket
    * @param s3Source AWS S3 credentials to access bucket and object
    */
-  case class S3FilePublished(path: String, s3Source: S3Source, observer: ActorRef) extends ObserverBatchEvent {
-    def streamContent = s3Source.s3.get(s3Source.bucket, path).map(_.content)
+  case class S3FilePublished(id: String, s3Source: S3Source, observer: ActorRef) extends ObserverFileEvent {
+    def streamContent: Option[InputStream] = s3Source.s3.get(s3Source.bucket, id).map(_.content)
+  }
+
+  /**
+   * Record has been received from AWS Kinesis Stream
+   *
+   * @param streamName name of kinesis stream
+   * @param seqNr      unique sequence number assigned to record
+   * @param data       buffer containing data carried by record
+   * @param observer   observer that witnessed the record receipt
+   */
+  case class KinesisRecordReceived(streamName: String, seqNr: String, data: ByteBuffer, observer: ActorRef) extends ObserverCommandEvent {
+    val byteBuffer = data
+    val byteArray = new Array[Byte](byteBuffer.remaining())
+    byteBuffer.get(byteArray)
```

Comment:
External contribution, but as far as I can tell this is just a way of converting a `ByteBuffer` to an input stream by copying it to a `ByteArray` since JVM has a built-in `ByteArrayInputStream` but no `ByteBufferInputStream`. It's a decent solution IMO since the alternative is using a custom `ByteBufferInputStream`.